src/domain.py

```diff
@@ -772,6 +774,19 @@ def _apply(
                     self.entity, next_settings["recovery_password"]
                 )
 
+            remove_auto_dns_feature = "use_auto_dns" in next_settings and not next_settings["use_auto_dns"]
+            if remove_auto_dns_feature:
```

Comment:
```suggestion
            # NB: this is subtlely different from just checking `next_settings.get("use_auto_dns") since we want to find the exact situation where the admin *disables* the autodns`
            remove_auto_dns_feature = "use_auto_dns" in next_settings and not next_settings["use_auto_dns"]
            if remove_auto_dns_feature:
```

Reply:
(Adding a comment - if i understood correctly the intent here)

Reply:
Indeed, we want to remove thing only when the value changes and is falsy

Reply:
yeah that's what i was trying to convey with my comment :sweat_smile: